src/lib.rs

```diff
@@ -322,7 +341,7 @@ pub fn primitive_enum(tokens: TokenStream) -> TokenStream {
         ident_token("repr"),
         paren_token(repr_type.clone()),
     ]));
-
+    check_for_default(&mut triples); // make sure there's a default, if the user didn't specify one
```

Comment:
Can you move this this call inside the block where `triples` is computed?

So instead of having

```rust
let mut triples = {
  let mut triples = ...
  ...
  triples
};
...
check_for_default(&mut triples);
```

I would prefer

```rust
let triples = {
  let mut triples = ...
  ...
  check_for_default(&mut triples);
  triples
};
```

IIUC, this will still have the same behavior, but this way we don't have to make `triples` mut.